WcaOnRails/app/webpacker/javascript/results/ResultForm/ResultForm.js

```diff
@@ -0,0 +1,214 @@
+  //   to identify it!)
+  //   - if we're creating a result, we want to let the backend take the
+  //   requested round attribute, and figure out if the data are appropriate
+  //   (ie: if it matches an existing round).
+  return { result: res, round };
+};
+
+const ResultForm = ({
+  result, saveAction, deleteAction, saving, response,
+}) => {
+  const solveCount = result.format_id
+    ? formats.byId[result.format_id].expectedSolveCount : 0;
+  const computeAverage = [3, 5].includes(solveCount) && result.event_id !== '333mbf';
+
+  // Person-related state
+  const [personData, setPersonData] = useState(personDataFromResult(result));
+
+  // Immutable round-related data
+  // FIXME: now that we have a 'Round' object for every round in the database
+  // (I think), we may consider showing its data here, which would make it
+  // slightly easier to select the appropriate round when creating a result,
+  // if at some point we want to make it selectable.
+  const [roundData, setRoundData] = useState(roundDataFromResult(result));
+
+  // Attempts-related state
+  const [attemptsState, setAttemptsState] = useState(attemptsDataFromResult(solveCount, result));
+
+  // Populate the original states whenever the original result changes.
+  useEffect(() => {
+    setAttemptsState(attemptsDataFromResult(solveCount, result));
+    setRoundData(roundDataFromResult(result));
+    setPersonData(personDataFromResult(result));
+  }, [result]);
+
+  return (
+    <div className="result-form">
+      <h3>
+        Round data
+      </h3>
+      <RoundForm roundData={roundData} />
+      <h3>
+        Person data
+      </h3>
+      <PersonForm personData={personData} setPersonData={setPersonData} />
+      <h3>
+        Attempts
+      </h3>
+      <AttemptsForm
+        eventId={result.event_id}
+        state={attemptsState}
+        setState={setAttemptsState}
+        computeAverage={computeAverage}
+      />
+      <SaveMessage response={response} />
+      <Button
+        positive
+        loading={saving}
+        disabled={saving}
+        onClick={() => saveAction(dataToResult(roundData, personData, attemptsState))}
+      >
+        Save
+      </Button>
+      {result.id && (
+        <DeleteResultButton deleteAction={() => deleteAction(result)} />
+      )}
+    </div>
+  );
+};
+
+// This is a simple wrapper to be able to manage request-specific states,
+// and to be able to hide the form upon creation.
+const ResultFormWrapper = ({ result, sync }) => {
+  const { id } = result;
+  const newResult = id === undefined;
+
+  // Saving and user-feedback states.
+  const { save, saving } = useSaveData({ method: newResult ? 'POST' : 'PATCH' });
+  const [response, setResponse] = useState({});
+
+  // This is used to track if we did save something.
+  const [saved, setSaved] = useState(undefined);
+
+  const [deleted, setDeleted] = useState(undefined);
+
+  const successAction = useCallback((data, responseJson) => {
```

Comment:
@saranshgrover this is the post-save handler which is called after a successful `save()` request (the `save` method is created a few lines above).

In the "edit result" form we display the state of the currently saved result in the database (additionally to the form itself), so we actually need to call the `sync` method if something was changed in the database.
If there was only the form I don't even think calling the `sync` method would be necessary, because a successful save would mean our current state is actually what is in the db.

It's worth noting that a successful request doesn't mean something has been saved: the controller may have encountered validation errors and filled the response with them.
If the response doesn't have errors, then we can call `sync` to reload the result saved in the db (see a couple of lines below).